lib/rubocop/cop/style/stderr_puts.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop identifies places where `$stderr.puts`
+      # can be replaced by `warn`.
+      #
+      # @example
+      #   # bad
+      #   $stderr.puts('hello')
+      #
+      #   # good
+      #   warn('hello')
+      #
+      class StderrPuts < Cop
+        MSG = 'Use `$stderr.puts` instead of `warn`.'.freeze
+
+        def_node_matcher :stderr_puts?, <<-PATTERN
+          (send
+            (gvar #stderr_gvar?) :puts
```

Comment:
Maybe this cop should add an offense for `STDERR.puts something`. if `$stderr` is re-assigned, `STDERR.puts something` has different mean. But in most cases, these have the same meaning.

Reply:
Your argument is correct. I think this should just be configurable - flag both by default, but have an option to flag just `STDERR`.

Reply:
I see. As for `STDERR.puts`, it seems good to separately implement cop according to the following guide and solve in combination. WDYT?
https://github.com/bbatsov/ruby-style-guide#global-stdout

Reply:
Sounds good.